frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/GitMacheteRepositoryReadyAction.java

```diff
@@ -19,13 +19,11 @@
   public void update(AnActionEvent anActionEvent) {
     super.update(anActionEvent);
 
-    var presentation = anActionEvent.getPresentation();
-    if (ActionUtils.getGitMacheteRepository(anActionEvent).isDefined()) {
-      presentation.setEnabled(true);
-      presentation.setVisible(true);
-    } else {
-      presentation.setEnabled(false);
-      presentation.setVisible(false);
+    boolean isEnabled = ActionUtils.getGitMacheteRepository(anActionEvent).isDefined();
+    anActionEvent.getPresentation().setEnabled(isEnabled);
```

Comment:
Not changing visibility anymore?

Reply:
nope :no_good: for the sake of UX we prefer disabled actions with clarifying description over vanishing actions. the only exceptions from it are root branch-related actions because of actions like `slide out` or `rebase onto parent` it is never possible. 

The logic is like that:
_If an action can be performed (satisfying some conditions!) for the given context* but it is not possible right now (cause the "some conditions" are not satisfied at this moment) then we diable it_

By "context" I mean graph table toolbar, root branch context menu, non root branch context menu (and possibly `Find Actions...` dialog!).

WDYT?

Reply:
Oh ok, I see the reasoning; you're right we shouldn't hide action by default (esp. given that this is kinda base class for most if not all other actions)

Reply:
> If an action can be performed...

Yeah, that's indeed a sane UX rule (at least on desktops, I think mobiles rarely display disabled options for the sake of space conservation)